### PR TITLE
Zone activity - farm plot tilling

### PR DESCRIPTION
**Pull request by Robik81**
#### Summary

```SUMMARY: Features "adds new Farm: Plot zone and an activity to till it"```

#### Purpose of change

As outlined in #24826 , this PR aims at reducing tedium of farming. Tilling is first activity for Farm: Plot zone, planting etc. will follow in separate PRs.

#### Describe the solution

1. Designate new zone of Farm: Plot type in Zone manager. 
2. Perform Zone activity - formerly Sort out loot, was renamed to Zone activities (shortcut Shift+O).
3. Choose Till farm plots, if more than one activity is available.
4. Watch your character do all the work automatically.

#### Additional context

Also contains code infrastructure for future fixing of #25116

### CODE REVIEW

**Comment by Robik81** [comment]
Damn, you people are on the roll. That was what, 50 PRs merged in last two days or something?

@Brambor Thank you, I sincerely hope the feature will serve well all aspiring farmers in this great game.

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -2781,3 +2783,77 @@ void activity_handlers::haircut_finish( player_activity *act, player *p ) {
...
+        const auto &tile_loc = g->m.getlocal( tile );
+
+        auto route = g->m.route( p->pos(), tile_loc, p->get_pathfinding_settings(), p->get_path_avoid() );
+        if( route.size() > 1 ) {
+            route.pop_back();
+            p->set_destination( route, player_activity( activity_id( "ACT_TILL_PLOT" ) ) );
+            return;
+        } else { // we are at destination already
+            p->add_msg_if_player( _( "You churn up the earth here." ) );
+            p->moves = -300;
```
Comment (ZhilkinSerg) [inline src/activity_handlers.cpp]: I believe this line should be:

`p->moves -= 300;` - i.e. subtract 300 from player moves.

Now it sets player moves to -300 which makes following check (`if( p->moves <= 0 ) {`) redundant as this expression will always be true.

Comment (Robik81) [reply src/activity_handlers.cpp]: Yeah, I think you are correct, that was my intended behavior... not that I came with the formula, I stole these 3 lines from iuse::makemound and I did not check them properly I guess. I think it should be changed in both places. I would normally not duplicate the code, but I did, as this was only 3 lines.

Should I make a PR to add a cost to game_constants.h and use the value in both places? I think it is better than making extra function with just 3 lines of code.

Comment (ZhilkinSerg) [reply src/activity_handlers.cpp]: No new function is necessary right now and you can simply change values in existing places.

Btw, you can also use following syntax: `p->mod_moves( -300 );`

Comment (Robik81) [reply src/activity_handlers.cpp]: Fix is in #25253 , tested both manual and automatic tilling.

**Pull request merged**